crates/node-e2e-tests/src/node.rs

```diff
@@ -0,0 +1,310 @@
+use alloy_rpc_types::BlockNumberOrTag;
+use eyre::Ok;
+use jsonrpsee::http_client::HttpClient;
+use reth::{
+    args::{DiscoveryArgs, NetworkArgs},
+    blockchain_tree::ShareableBlockchainTree,
+    builder::{
+        components::FullNodeComponentsAdapter, FullNode, FullNodeTypesAdapter, NodeBuilder,
+        NodeHandle,
+    },
+    network::{NetworkEvent, NetworkEvents, NetworkHandle, PeersInfo},
+    providers::{
+        providers::BlockchainProvider, BlockReaderIdExt, CanonStateNotificationStream,
+        CanonStateSubscriptions,
+    },
+    revm::EvmProcessorFactory,
+    rpc::{
+        api::EngineApiClient,
+        eth::{error::EthResult, EthTransactions},
+        types::engine::{ExecutionPayloadEnvelopeV3, ForkchoiceState, PayloadAttributes},
+    },
+    tasks::TaskExecutor,
+    transaction_pool::{
+        blobstore::DiskFileBlobStore, CoinbaseTipOrdering, EthPooledTransaction,
+        EthTransactionValidator, Pool, TransactionValidationTaskExecutor,
+    },
+};
+use reth_db::{test_utils::TempDatabase, DatabaseEnv};
+use reth_node_core::{args::RpcServerArgs, node_config::NodeConfig};
+use reth_node_ethereum::{EthEngineTypes, EthEvmConfig, EthereumNode};
+use reth_payload_builder::{
+    EthBuiltPayload, EthPayloadBuilderAttributes, Events, PayloadBuilderHandle, PayloadId,
+};
+use reth_primitives::{Address, Bytes, ChainSpec, NodeRecord, B256};
+use reth_tracing::tracing::info;
+use std::{
+    sync::Arc,
+    time::{SystemTime, UNIX_EPOCH},
+};
+use tokio_stream::{
+    wrappers::{BroadcastStream, UnboundedReceiverStream},
+    StreamExt,
+};
+
+type TestProvider = BlockchainProvider<
+    Arc<TempDatabase<DatabaseEnv>>,
+    ShareableBlockchainTree<Arc<TempDatabase<DatabaseEnv>>, EvmProcessorFactory<EthEvmConfig>>,
+>;
+
+/// An helper struct to handle node actions
+pub struct NodeHelper {
+    engine_api_client: HttpClient,
+    canonical_stream: CanonStateNotificationStream,
+
+    payload_event_stream: BroadcastStream<Events<EthEngineTypes>>,
+    payload_builder: PayloadBuilderHandle<EthEngineTypes>,
+
+    network_events: UnboundedReceiverStream<NetworkEvent>,
+    network: NetworkHandle,
+
+    provider: TestProvider,
+
+    node: TestNode,
```

Comment:
I think we want a wrapper type over

https://github.com/paradigmxyz/reth/blob/dc9fc372cb257f47d7636b25559cdaffa65003e9/crates/node-builder/src/node.rs#L44-L44

here